### PR TITLE
UPSTREAM: 76189: Fix flaky legacy pod autoscaler test

### PR DESCRIPTION
**Pull request by soltysh**

### CODE REVIEW

File: vendor/k8s.io/kubernetes/pkg/controller/podautoscaler/legacy_horizontal_test.go [hunk 1]
```diff
@@ -472,7 +472,10 @@ func (tc *legacyTestCase) runTest(t *testing.T) {
 		if tc.finished {
 			return true, &v1.Event{}, nil
 		}
-		obj := action.(core.CreateAction).GetObject().(*v1.Event)
+		obj, ok := action.(core.CreateAction).GetObject().(*v1.Event)
```
Comment (ingvagabund) [inline vendor/k8s.io/kubernetes/pkg/controller/podautoscaler/legacy_horizontal_test.go]: So the client-go code says the same event (without defining what `the same` means) is patched when it's has been already recorded [1]. The fix would be then to allow to accept even Patch events. Hoping the Reason will stay the same. Or, just ignore any patch event.

[1] https://github.com/kubernetes/client-go/blob/master/tools/record/event.go#L214-L216

**Comment by soltysh** [comment]
That unit flake is a separate one.
/retest

File: vendor/k8s.io/kubernetes/pkg/controller/podautoscaler/legacy_horizontal_test.go [hunk 2]
```diff
@@ -472,7 +472,11 @@ func (tc *legacyTestCase) runTest(t *testing.T) {
 		if tc.finished {
 			return true, &v1.Event{}, nil
 		}
-		obj := action.(core.CreateAction).GetObject().(*v1.Event)
+		create, ok := action.(core.CreateAction)
+		if !ok {
+			return false, nil, nil
```
Comment (ingvagabund) [inline vendor/k8s.io/kubernetes/pkg/controller/podautoscaler/legacy_horizontal_test.go]: Per
```
// ReactionFunc is a function that returns an object or error for a given
// Action.  If "handled" is false, then the test client will ignore the
// results and continue to the next ReactionFunc. 
```
lgtm. Just ignoring every patch operation.

**Comment by smarterclayton** [comment]
Force merging to unblock the queue (all tests passed once)

**Pull request merged**